### PR TITLE
6144 js logger in db

### PR DESCRIPTION
**Pull request by mmpei**
test cases:

1. Start a image scan job. Job successful. after job finished, check data in table Job_log in DB.
2. Start a image scan job. Job failed. after job finished, check data in table Job_log in DB.
3. Check the Job log produced in step 1 and 2, Log from step 2 checked from UI and 1 from OpenAPI.
4. Confing using sweeper，use default Duration(1 day)，Start JobService，the expired logs will be clean immediately.
5. Confing using sweeper，use default Duration(1 day)，Start JobService，the expired logs will be clean after one day.
6. delete DB data, and startup harbor again,  DB will be created and table job_log will be created too.
7. Case for limit of length of Job content. when size of log proceeds 4k retrieving log normally. No limit for log size.
8. all Unit Test cases run successfully。

### CODE REVIEW

File: src/common/dao/joblog.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package dao
+
+import (
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/astaxie/beego/orm"
+	"time"
+)
+
+// CreateOrUpdateJobLog ...
+func CreateOrUpdateJobLog(log *models.JobLog) (int64, error) {
```
Comment (reasonerjt) [inline src/common/dao/joblog.go]: Does it need to return count?

File: src/jobservice/job/impl/context.go [hunk 2]
```diff
@@ -95,7 +96,14 @@ func (c *Context) Init() error {
...
 	db := getDBFromConfig(configs)
 
-	return dao.InitDatabase(db)
+	err = dao.InitDatabase(db)
+	if err != nil {
+		return err
+	}
+
+	// Initialize DB finished
+	initDBCompleted()
```
Comment (reasonerjt) [inline src/jobservice/job/impl/context.go]: Is it necessary?  IMO we can assume the DB is ready when DBSweeper sweeps the log?

File: src/jobservice/logger/sweeper/db_sweeper.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	}
+}
+
+// Sweep logs
+func (dbs *DBSweeper) Sweep() (int, error) {
+	// DB initialization not completed, waiting
+	WaitingDBInit()
+
+	// Start to sweep logs
+	before := time.Now().Add(time.Duration(dbs.duration) * oneDay * -1)
```
Comment (reasonerjt) [inline src/jobservice/logger/sweeper/db_sweeper.go]: time.Now().Sub?

File: src/jobservice/logger/known_loggers.go [hunk 4]
```diff
@@ -79,3 +87,24 @@ func IsKnownLevel(level string) bool {
 
 	return false
 }
+
+// Return a logger name by Interface
+func GetLoggerName(l Interface) string {
```
Comment (reasonerjt) [inline src/jobservice/logger/known_loggers.go]: Maybe we should add `func Name() string` to logger interface? @steven-zou

File: src/jobservice/job/impl/context.go (hunk 2 above)
Comment (mmpei) [reply src/jobservice/job/impl/context.go]: actually not, DB initialized after logger and DB initialization relay on logger.
Sweeper will be executed once as soon as logger initialization finished.

File: src/jobservice/logger/sweeper/db_sweeper.go (hunk 3 above)
Comment (mmpei) [reply src/jobservice/logger/sweeper/db_sweeper.go]: sub accept  a parameter Time and return Duration, not applicable.

File: src/common/dao/joblog.go (hunk 1 above)
Comment (mmpei) [reply src/common/dao/joblog.go]: not necessary. always return 1 for Job_uuid is a unique index in DB.
just a result from ORM InsertOrUpdate.

File: src/jobservice/logger/sweeper/db_sweeper.go (hunk 3 above)
Comment (reasonerjt) [reply src/jobservice/logger/sweeper/db_sweeper.go]: correct, my bad

**Comment by steven-zou** [comment]
@mmpei 

Thanks a lot for such great PR. I did not see the decrease count `-0.2%` before merging. It is a little bigger (We have merging criteria, the coverage should not decrease more than `-%0.03`)  Would u please to add more UT cases to increase the testing coverage in a new PR?

**Comment by mmpei** [comment]
OK, i will add some UT code and create a new PR these days.

**Pull request merged**